src/submission/validation-clinical/specimen.ts

```diff
@@ -28,6 +28,13 @@ export const validate = async (
 
   const errors: SubmissionValidationError[] = []; // all errors for record
 
+  const specimen = getSpecimenFromDonor(existentDonor, specimenRecord, errors);
+  if (!specimen) {
+    return errors;
+  }
+
+  checkRequiredFields(specimen, specimenRecord, errors);
```

Comment:
is there a unit test to cover this ?

Reply:
Unit test can be done after merge as this will go into QA testing to save sometime.